pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -461,55 +465,65 @@ func (h *upgradeHandler) cleanup(upgrade *harvesterv1.Upgrade, cleanJobs bool) e
 			}
 			jobs, err := h.jobCache.List(plan.Namespace, set.AsSelector())
 			if err != nil {
-				return err
+				return nil, err
 			}
 			for _, job := range jobs {
 				logrus.Debugf("Deleting job %s/%s", job.Namespace, job.Name)
 				if err := h.jobClient.Delete(job.Namespace, job.Name, &metav1.DeleteOptions{}); err != nil {
-					return err
+					return nil, err
 				}
 			}
 		}
 
 		logrus.Debugf("Deleting plan %s/%s", plan.Namespace, plan.Name)
 		if err := h.planClient.Delete(plan.Namespace, plan.Name, &metav1.DeleteOptions{}); err != nil {
-			return err
+			return nil, err
 		}
 	}
 
 	// restore Longhorn replica-replenishment-wait-interval and
 	// auto-cleanup-system-generated-snapshot settings (multi-node cluster only)
 	if upgrade.Status.SingleNode == "" {
 		if err := h.loadReplicaReplenishmentFromUpgradeAnnotation(upgrade); err != nil {
-			return err
+			return nil, err
 		}
 		if err := h.loadAutoCleanupSystemGeneratedSnapshotFromUpgradeAnnotation(upgrade); err != nil {
-			return err
+			return nil, err
 		}
 	}
 
 	// tear down logging infra if any
 	if harvesterv1.LogReady.IsTrue(upgrade) && upgrade.Status.UpgradeLog != "" {
 		upgradeLog, err := h.upgradeLogCache.Get(upgradeNamespace, upgrade.Status.UpgradeLog)
 		if err != nil {
-			if apierrors.IsNotFound(err) {
-				return nil
```

Comment:
Previously, we have this early return when upgradeLog CR couldn't be fetched. I think it is a little bit weird since this would skip the `h.resumeManagedCharts()` part below.

So I remove the early return part in this PR.

Reply:
@w13915984028 Got it, thanks for the suggestion. I'll revise the PR as your suggestion to limit the change scope.

Reply:
There is one https://github.com/harvester/harvester/issues/8442, but is not dedicatedly for the upgrade controller state machine refactor when it was filed.